From f8e600240d95a46ad810a2b2cc20c41f4c135887 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Thu, 11 May 2017 14:05:20 -0700
Subject: [PATCH] utils: add support to load/save preferences in
 XDG_CONFIG_HOME

Preferences will be loaded/saved there when set.
---
 src/utils.cc | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/src/utils.cc b/src/utils.cc
index ef41f67..b21175e 100644
--- a/src/utils.cc
+++ b/src/utils.cc
@@ -302,8 +302,13 @@ bool LoadPref(void)
   
 #if defined(LINUX) && !defined(__AMIGAOS4__)
   char PathPref[512];
-  char *Env=getenv("HOME");
-  sprintf(PathPref,"%s/.ri-li.pref",Env);
+  char *Env=getenv("XDG_CONFIG_HOME");
+  if (Env) {
+    sprintf(PathPref,"%s/ri-li.pref",Env);
+  } else {
+    Env=getenv("HOME");
+    sprintf(PathPref,"%s/.ri-li.pref",Env);
+  }
 #elif defined(__AMIGAOS4__)
   char PathPref[512];
   char *Env="PROGDIR:";
@@ -339,8 +344,13 @@ void SauvePref(void)
 {  
 #if defined(LINUX) && !defined(__AMIGAOS4__)
   char Provi[512];
-  char *Env=getenv("HOME");
-  sprintf(Provi,"%s/.ri-li.pref",Env);
+  char *Env=getenv("XDG_CONFIG_HOME");
+  if (Env) {
+    sprintf(Provi,"%s/ri-li.pref",Env);
+  } else {
+    Env=getenv("HOME");
+    sprintf(Provi,"%s/.ri-li.pref",Env);
+  }
   SauveFichier(Provi,(char*)&Pref,sizeof(sPreference));
 #elif defined(__AMIGAOS4__)
   char Provi[512];
-- 
2.12.2

